### DISCUSSION TITLE
Tooltip with Tree component

### DISCUSSION THREAD

**Discussion by MartynasStrazdas** [iTwin/iTwinUI#2342] [Q&A]
Recently we started adding missing tooltips for our Tree implementation and have found some issues.

1. When deselecting a tree node with a tooltip, the tooltip may overlap and block other nearby nodes, making navigation difficult. Simply removing the tooltip when the mouse leaves the trigger element as I understand would harm accessibility. What improvements can be made to enhance mouse navigation in this situation?

https://github.com/user-attachments/assets/fe1b201a-6dc1-4859-9eab-c96eaea2d8ad

2. Expander for tree node does not have a property for title/label or a way to set a tooltip for the default implementation. Could it be possible to add an option for tooltip message in `TreeNodeExpanderProps`

**Comment by r100-stack** [comment]
Thanks for reporting this. 

1. Before beginning, just wanted to mention that Tooltips should not be used for revealing truncated text ([more info](https://itwinui.bentley.com/docs/tooltip#best-practices)). So, I don't believe the Tooltip should be used here since iiuc, the tooltip is revealing truncated text. We can instead discuss ways to avoid truncation in the UI, such as text wrapping.
    - If the actual Tooltip issue you pointed out happens in other cases where all Tooltip pre-requisite requirements are met, you could explore the `Tooltip`'s `placement` prop to move the Tooltip out of the way of where the mouse would likely be. E.g. we did this in the `TransferList.Toolbar`. You can see the screen-recordings in the PR description of ([#2274](https://github.com/iTwin/iTwinUI/pull/2274)).

2. Not sure I understand this correctly. Because we do offer the `label` prop via `TreeNode`'s `expanderProps` (`TreeNodeExpanderProps`) prop. We already add an `aria-label` of "Expand" to the expand button. But you can override it with the `label`/`aria-label`/`aria-labeledby` prop.

**Reply by grigasp** [reply]
> We can instead discuss ways to avoid truncation in the UI, such as text wrapping.

Our problem consists of:
- narrow tree component,
- deep hierarchies,
- nodes with long labels,
- nodes having extra stuff in front (checkboxes) and back (action buttons).

A combination of all of that can easily make just a few label letters visible. So we'd like some way for users to see the labels, especially when navigating through them with keyboard. But even without keyboard navigation, being able to see them without needing to scroll horizontally would be great.

If tooltips aren't the solution, then we'd be thankful for any advice.
